src/core/types/Vote.ts

```diff
@@ -0,0 +1,17 @@
+import { Address } from './Common';
+
+export type Vote = {
+  address: Address;
+  weight: number;
+  timestamp: number; // We can get the vote time from onchain. Not sure if there is use for it
+}
+
+export type YesNoVoteOption = 'YES' | 'NO' | 'ABSTAIN';
```

Comment:
I wonder if it's worth it to completely generalise vote options (on the DB level or close to it) so that as far as the API is concerned, there are: a proposal, a number of possible voting options on the proposal and the weights on each option

Reply:
Yes maybe all votes are multiple choice and indexed from `voteChoices[]`

Reply:
https://linear.app/decent-labs/issue/ENG-761/generalize-vote-type-options